### DISCUSSION TITLE
Another noob with questions :)

### DISCUSSION THREAD

**Discussion by iawa2k** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#600] [Q&A]
I have just got my device setup and connected to my 6 wire 2021Maldives (S200102) with jets using the latest PCB (V2B), red level shifter, JST cables and the NodeMCU V3 ESP8266 which works a dream.

Even setup Mosquito on my Homebridge server and added the config in as well as Jets and all working through HomeKit perfectly.

I am not running the beta but the latest Master build (2023-08-16-1957).

I have a couple of questions;

1. The sound the hot tub makes is different, I saw references to being able to customise this but not sure how? Is it only for beta atm?

2. I may be just not understanding it but how do I calibrate the virtual temperature? I can't select the tickbox?

3. The "Ready In" shows "-1d -1:00:00 (Never)". Is this to do with calibration? I saw on another post the Ambient Temp needs setting, I don't use HA (just HomeBridge) and don't have any external temp sensors (though could get one I guess, but not keen on running HA). I set the Ambient Temperature using the interface, does that not do anything for the "Ready In"? Why is it there if not?

4. Last one, promise! When my tub gets up to temp the pump and heater now turn off, is that normal? Does it automatically come on when it thinks the "Virtual Temp" is too low to bring it back up or is that just something that can happen?

Otherwise, great project and am certainly leaving a coffee tip as can see so much work and effort has gone into this fine project!

**Comment by visualapproach** [comment]
> 1. The sound the hot tub makes is different, I saw references to being able to customise this but not sure how? Is it only for beta atm?

The audio is customized by me and cannot easily be changed by user. It's possible that I will make it more flexible in the future.

> 2. I may be just not understanding it but how do I calibrate the virtual temperature? I can't select the tickbox?

The tickbox is checked by the device when it believes it calibrated. Mouseover the question marks to get more info. You can set it manually also.

> 3. The "Ready In" shows "-1d -1:00:00 (Never)". Is this to do with calibration? I saw on another post the Ambient Temp needs setting, I don't use HA (just HomeBridge) and don't have any external temp sensors (though could get one I guess, but not keen on running HA). I set the Ambient Temperature using the interface, does that not do anything for the "Ready In"? Why is it there if not?

Yes, when rate of cooling is larger than the heating capacity it will never reach target temperature. The ambient temp can be used from the UI which will affect the estimated heating time. However it's intended to be fed automatically from some home automation system. Not necessarily HA. I don't know much about homebridge but if it can get temp from met offices it would suffice.
Remember to calibrate or the estimated time and virtual temp will be wrong.

> 4. Last one, promise! When my tub gets up to temp the pump and heater now turn off, is that normal? Does it automatically come on when it thinks the "Virtual Temp" is too low to bring it back up or is that just something that can happen?

Unless you added items in the command queue or having automations made from HB the pump should stay on even when the temp is reached. The heater button should also be on, but turn green. (On the display)
It should behave like factory settings.

Thank you very much for the coffee! Helps to get me going ☕

**Comment by iawa2k** [comment]
1. Thank you, wasn’t sure if something I should be seeing.

2. Ah I see, it’s an automatic process, I’ll leave it tonight with the pump on and no heater and hopefully that should resolve.

3. So until it’s calibrated it will think the rate of cooling is higher than the heating capacity, correct? Happy to set ambient temp manually for now and I’ve seen a PowerShell module (M2MQTT) that I can probably script to update the temp from Met Office or similar source on a schedule.

4. I haven’t but saying that it hasn’t seemed to do it since my post so maybe it was just a settling in thing, I literally powered it up inside the device about 2 hours before I posted 👍 


You’re most welcome regarding coffee, it’s a great project and was very easy to follow along and get correct bits ordered etc.

**Comment by iawa2k** [comment]
As a follow-up, I ended up writing a PowerShell script to run every so often to get the current temperature from OpenWeatherMaps.org who have a free API.

PowerShell Code:

_Add-Type -Path "C:\lib\M2Mqtt.4.3.0.0\lib\net45\M2Mqtt.Net.dll"

function global:MQTTMsgReceived {
    Param(
        [parameter(Mandatory=$true)]$mqtt
    )
    $topic = $mqtt.topic
    $msg = $([System.Text.Encoding]::ASCII.GetString($mqtt.Message))
    Write-Host ("Topic: " + $topic)
    Write-Host ("Message: " + $msg)
}

‘# Fetch weather data from the API (OpenWeatherMap example)
$apiKey = "YOUR OPENWEATHERMAP.ORG API KEY"
$lat = "YOUR LATITUDE" #Example Latitude = Croydon, UK is 51.3833
$lon = "YOUR LONGITUDE"    #Example Longitude = Croydon, UK is -0.1
$apiUrl = "http://api.openweathermap.org/data/2.5/weather?lat=$lat&lon=$lon&appid=$apiKey&units=metric"
$response = Invoke-RestMethod -Uri $apiUrl -Method Get
$tempinC = $response.main.temp

‘# Connect to MQTT broker
$MqttBroker = "YOUR MQTT BROKER IP"
$MqttClient = New-Object uPLibrary.Networking.M2Mqtt.MqttClient($MqttBroker)
$MqttClient.Connect([guid]::NewGuid().ToString(), "layzspa", "1wtwHTT")

‘# Construct JSON payload
$payload = @{
    CMD = 15
    VALUE = [math]::Round($tempinC)
    XTIME = 0
    INTERVAL = 0
} | ConvertTo-Json

‘#Publish message
$MqttClient.Publish("spamqtt/command", [System.Text.Encoding]::UTF8.GetBytes($payload), 0, $false)_


I Installed M2Mqqt module for plugin with nuget, guide at this site: https://jackgruber.github.io/2019-06-05-ps-mqtt/

Hopefully that helps someone else too!

**Reply by iawa2k** [reply]
Will need the ‘ quote removed from the commented out parts starting with # as it seems to try and make these heading formatted in the post lol.

**Reply by visualapproach** [reply]
referenced in manual
